### DISCUSSION TITLE
Creating Layouts and Viewports

### DISCUSSION THREAD

**Discussion by grantdtaylor** [DomCR/ACadSharp#387] [Q&A] [answered]
My question is how do I go about adding a new layout to a document and then add a viewport to the layout?

```
        CadDocument doc = new CadDocument();
        var layout = new ACadSharp.Objects.Layout("NewLayout01");
        doc.Layouts.Add(layout);

        using (DwgWriter writer = new DwgWriter(outPath, doc))
        {
            writer.OnNotification += NotificationHelper.LogConsoleNotification;
            writer.Write(); // throws System.ArgumentNullException: 'Value cannot be null. Arg_ParamName_Name'
        }
```

Many thanks,
Grant

**Comment by DomCR** [comment, marked as answer]
Hi @grantdtaylor,

Right now you cannot create a new `Layout`, I haven't look enough at the functionality of them so the feature is blocked to ensure stability in the project.

I'll open an Issue and a PR to add this feature to the project.
